packages/mdx/src/index.ts

```diff
@@ -156,12 +158,29 @@ export const Themed: ThemedDiv & ThemedComponentsDict = styled('div')(
   themed('div')
 ) as ThemedDiv & ThemedComponentsDict
 
+// to be deprecated
+export const Styled: ThemedDiv & ThemedComponentsDict = styled('div')(
+  themed('div')
+) as ThemedDiv & ThemedComponentsDict
+
+export const warnStyled = (tag: keyof IntrinsicSxElements): FC => (props) => {
+  useEffect(() => {
+    if (process.env.NODE_ENV !== 'production') {
+      console.warn(
+        'The Styled component from "@theme-ui/mdx" is deprecated and will be removed in a future version, please use Themed instead'
+      )
+    }
+  }, [])
+  return createElement(tag, props)
+}
+
 export const components = {} as ThemedComponentsDict
 
 tags.forEach((tag) => {
   // fixme?
   components[tag] = styled(alias(tag))(themed(tag)) as any
   Themed[tag] = components[tag] as any
+  Styled[tag] = styled(warnStyled(tag))(themed(tag)) as any
```

Comment:
What do you think about using a proxy instead of copying the code for Themed?

I've been thinking about something like this: [**TS Playground**](https://www.typescriptlang.org/play?#code/MYewdgzgLgBADgJxMAphCMC8MDeMVgBuAXLgL4wCGGASiqAgCYA80CAlmAOYA0MArmADWYEAHcwAPhhkA3AFgAUEtCRYXBChRQUCLLhgRKATwASKADYWQACgCUB1RBAWUAOmtcbAcnNWQMADqIAgWjACE3g4UZEoq4NAwjCiI9JQ6jADimtq6+mAoYjAACkgAHsY2Glo6CHw4SjBNMFzaNlCUCK1QfEIoxg4Nis0jMOwAZjA2iMhoEG4EhG4AcgDyACIAogD6m8sAajDhmNjeM4z8wFDs4FG4jaOPE1N9xmNgMB1d2oMPj-8wJwudxiTpgGx-AFQgAGAGUoMZXIw3AASHDwjjcGyvOwUcZIAC2MHxICJACIAAJQAAWKAJKAAtPx2AB6AmMMpksYYZKpYDpFCMKhgIVidhWGAAIxQME0BJAhEF72hkKh-wA1DBoVRifwoPxNDBFQgIDcwHw4K5qDL+BAZQAVWn05FojGcLw4iicaAoSjI6F2VX-WLDYNxUOPTT6hAfL7dADaryoGFeIEmCLgKDTn063QAugoIzIlGRA8pFNVcgg3EYzJZrPYlLzNPyMtkaroayY-A27LIgA)

```ts
// after rename
const greeter = { sayHello() { console.log('Hello World!') } }

// before rename (like our Styled)
const deprecatedGreeter = new Proxy(greeter, {
    get(target, key) {
        if (process.env.NODE_ENV !== 'production') {
            if (key in target) {
                console.warn(
                    `Styled.${String(key)} from from "@theme-ui/mdx" is deprecated and will be removed in`
                    + ` a future version, please use Themed.${String(key)} instead.`)
            }
        }

        return target[key as keyof typeof target];
    }
})
```

Reply:
I would prefer to keep the old implementation - and avoid any new bugs.

Reply:
```
styled(warnStyled(tag))(themed(tag))
```

is a _new_ implementation. And it's bug-prone — right now `alias(tag)` is missing. Any accidental change can break it which won't get caught by Themed tests. 

![image](https://user-images.githubusercontent.com/15332326/105917427-94710c00-6032-11eb-98e3-28e95683188d.png)

This is a new bug. 

The approach I'm advising is 
1. Themed is the new API. We should thoroughly test that it works.
2. Styled is just an alias/proxy. We should briefly test that it's an alias/proxy. We should also test that it prints the warning.

Reply:
1. Regarding using a Proxy, I think we can use it for the new implementation of `Themed` - but keep `Styled` as close as possible to the old `Styled`. 

2. Great catch on alias - I am adding the aliased tags as well as a test for inlineCode.

3. There was already a test for `Styled`, just to make sure it functions. I am now adding tests for alias and console. warning

Reply:
> I assume that if we're deprecating something, we'll keep it to next stable "major" release.

agreed, I assume we are heading for the major release soon :)